src/test/integration/org/apache/cassandra/sidecar/routes/ClientStatsHandlerIntegrationTest.java

```diff
@@ -0,0 +1,170 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.sidecar.routes;
+
+import java.util.Collections;
+import java.util.Map;
+
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import io.vertx.core.buffer.Buffer;
+import io.vertx.ext.web.client.HttpResponse;
+import io.vertx.ext.web.client.predicate.ResponsePredicate;
+import io.vertx.junit5.VertxExtension;
+import io.vertx.junit5.VertxTestContext;
+import org.apache.cassandra.sidecar.common.response.ClientStatsResponse;
+import org.apache.cassandra.sidecar.testing.IntegrationTestBase;
+import org.apache.cassandra.testing.CassandraIntegrationTest;
+import org.apache.cassandra.testing.CassandraTestContext;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Test the client-stats endpoint with cassandra container.
+ */
+@ExtendWith(VertxExtension.class)
+public class ClientStatsHandlerIntegrationTest extends IntegrationTestBase
+{
+
+    @CassandraIntegrationTest
+    void retrieveClientStatsDefault(VertxTestContext context, CassandraTestContext cassandraTestContext)
+    throws Exception
+    {
+        String testRoute = "/api/v1/storage/client-stats";
+        testWithClient(context, client -> {
+            client.get(server.actualPort(), "127.0.0.1", testRoute)
+                  .expect(ResponsePredicate.SC_OK)
+                  .send(context.succeeding(response -> {
+                      assertClientStatsResponse(response, Collections.emptyMap(), cassandraTestContext.version.toString());
+                      context.completeNow();
+                  }));
+        });
+    }
+
+    @CassandraIntegrationTest
+    void retrieveClientStatsWithClientOptions(VertxTestContext context, CassandraTestContext cassandraTestContext)
+    throws Exception
+    {
+        Map<String, Boolean> params = Collections.singletonMap("client-options", true);
+        String testRoute = "/api/v1/storage/client-stats?client-options=true";
+        testWithClient(context, client -> {
+            client.get(server.actualPort(), "127.0.0.1", testRoute)
+                  .expect(ResponsePredicate.SC_OK)
+                  .send(context.succeeding(response -> {
+                      assertClientStatsResponse(response, params, cassandraTestContext.version.toString());
+                      context.completeNow();
+                  }));
+        });
+    }
+
+    @CassandraIntegrationTest
+    void retrieveClientStatsByProtocol(VertxTestContext context, CassandraTestContext cassandraTestContext)
+    throws Exception
+    {
+        Map<String, Boolean> params = Collections.singletonMap("by-protocol", true);
+        String testRoute = "/api/v1/storage/client-stats?by-protocol=true";
+        testWithClient(context, client -> {
+            client.get(server.actualPort(), "127.0.0.1", testRoute)
```

Comment:
Good point. However, since all the existing integration tests are defined this way, and the `Server` implementation does not expose the host API, I'd like to keep this as-is to keep the scope of this PR confined to the changes related to this API.